src/miral/application_selector.cpp

```diff
@@ -0,0 +1,71 @@
+//
+// Created by matthew on 7/18/23.
+//
+
+#include "application_selector.h"
+#include <mir/scene/session.h>
+
+using namespace miral;
+
+ApplicationSelector::ApplicationSelector(const miral::WindowManagerTools& in_tools)
+    : tools{in_tools}
+{}
+
+void ApplicationSelector::start(bool in_reverse)
+{
+    if (is_active())
+    {
+        return;
+    }
+
+    printf("Started\n");
```

Comment:
Debug code I presume - it should be removed. (Note you probably want to `flush(stdout)` too, as otherwise it can get confusing)